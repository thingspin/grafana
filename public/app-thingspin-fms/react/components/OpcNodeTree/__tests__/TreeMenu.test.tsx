import React from 'react';
import { shallow } from 'enzyme';

import TreeViewMenu from '../index';

const mockData = {
  atd: {
    label: 'ATS Guide',
    key: 'ats',
    index: 1, // ATS Guide should be after Release Notes
  },
  releasenotes: {
    label: 'Release Notes',
    key: 'releasenotes',
    index: 0, // Release Notes should be first
    nodes: {
      'desktop-modeler': {
        label: 'Desktop Modeler',
        key: 'releasenotes/desktop-modeler',
        index: 0,
        nodes: {
          7: {
            label: '7',
            key: 'releasenotes/desktop-modeler/7',
            index: 0,
            nodes: {
              '7.0': {
                label: '7.0',
                key: 'releasenotes/desktop-modeler/7.0',
                index: 0,
              },
            },
          },
        },
      },
    },
  },
};

describe('TreeViewMenu', () => {
  it('should render the level-1 nodes by default', () => {
    const wrapper = shallow(<TreeViewMenu data={mockData} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('li').length).toEqual(2);
  });
  it('should open specified nodes', () => {
    const wrapper = shallow(
      <TreeViewMenu
        data={mockData}
        openNodes={['releasenotes', 'releasenotes/desktop-modeler']}
      />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('li').length).toEqual(4);
  });
  it('should highlight the active node', () => {
    const activeKey = 'releasenotes/desktop-modeler/7';
    const highLightColor = 'white';
    const wrapper = shallow(
      <TreeViewMenu
        data={mockData}
        activeKey={activeKey}
        openNodes={['releasenotes', 'releasenotes/desktop-modeler']}
      />
    );
    const highlightedElement = wrapper.findWhere(node => node.key() === activeKey).get(0);

    expect(wrapper).toMatchSnapshot();
    expect(highlightedElement.props.style.color).toEqual(highLightColor);
  });
  it('should trigger onClickItem when a node is clicked', () => {
    const mockOnClickItem = jest.fn();
    const wrapper = shallow(
      <TreeViewMenu data={mockData} onClickItem={mockOnClickItem} />
    );

    const targetNode = wrapper.findWhere(node => node.key() === 'releasenotes');
    targetNode.simulate('click');
    expect(mockOnClickItem.mock.calls.length).toEqual(1);
    expect(mockOnClickItem).toHaveBeenCalledWith({
      hasNodes: true,
      index: 0,
      isOpen: false,
      key: 'releasenotes',
      label: 'Release Notes',
      level: 0,
      openNodes: [],
      parent: '',
      searchTerm: '',
    });
  });
});
