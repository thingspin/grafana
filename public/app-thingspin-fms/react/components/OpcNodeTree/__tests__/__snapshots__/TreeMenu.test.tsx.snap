// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeViewMenu should highlight the active node 1`] = `
<Fragment>
  <input
    aria-label="Type and search"
    onChange={[Function]}
    placeholder="Type and search"
    style={
      Object {
        "border": "none",
        "padding": "1rem 2rem",
        "width": "100%",
      }
    }
    type="search"
  />
  <ul
    style={
      Object {
        "borderTop": "1px solid #ccc",
        "listStyleType": "none",
        "paddingLeft": 0,
      }
    }
  >
    <li
      aria-pressed={false}
      key="releasenotes"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "none",
          "borderBottom": "1px solid #ccc",
          "color": "#333",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 3.5rem",
        }
      }
    >
      <div
        style={
          Object {
            "height": "2rem",
            "left": "2rem",
            "position": "absolute",
            "width": "2rem",
          }
        }
      >
        <ToggleIcon
          on={true}
        />
      </div>
      Release Notes
    </li>
    <li
      aria-pressed={false}
      key="releasenotes/desktop-modeler"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "none",
          "borderBottom": "1px solid #ccc",
          "color": "#333",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 4.75rem",
        }
      }
    >
      <div
        style={
          Object {
            "height": "2rem",
            "left": "3.25rem",
            "position": "absolute",
            "width": "2rem",
          }
        }
      >
        <ToggleIcon
          on={true}
        />
      </div>
      Desktop Modeler
    </li>
    <li
      aria-pressed={true}
      key="releasenotes/desktop-modeler/7"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "#179ed3",
          "borderBottom": "1px solid #ccc",
          "color": "white",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 6rem",
        }
      }
    >
      <div
        style={
          Object {
            "height": "2rem",
            "left": "4.5rem",
            "position": "absolute",
            "width": "2rem",
          }
        }
      >
        <ToggleIcon
          on={false}
        />
      </div>
      7
    </li>
    <li
      aria-pressed={false}
      key="atd"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "none",
          "borderBottom": "1px solid #ccc",
          "color": "#333",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 3.5rem",
        }
      }
    >
      ATS Guide
    </li>
  </ul>
</Fragment>
`;

exports[`TreeViewMenu should open specified nodes 1`] = `
<Fragment>
  <input
    aria-label="Type and search"
    onChange={[Function]}
    placeholder="Type and search"
    style={
      Object {
        "border": "none",
        "padding": "1rem 2rem",
        "width": "100%",
      }
    }
    type="search"
  />
  <ul
    style={
      Object {
        "borderTop": "1px solid #ccc",
        "listStyleType": "none",
        "paddingLeft": 0,
      }
    }
  >
    <li
      aria-pressed={false}
      key="releasenotes"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "none",
          "borderBottom": "1px solid #ccc",
          "color": "#333",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 3.5rem",
        }
      }
    >
      <div
        style={
          Object {
            "height": "2rem",
            "left": "2rem",
            "position": "absolute",
            "width": "2rem",
          }
        }
      >
        <ToggleIcon
          on={true}
        />
      </div>
      Release Notes
    </li>
    <li
      aria-pressed={false}
      key="releasenotes/desktop-modeler"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "none",
          "borderBottom": "1px solid #ccc",
          "color": "#333",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 4.75rem",
        }
      }
    >
      <div
        style={
          Object {
            "height": "2rem",
            "left": "3.25rem",
            "position": "absolute",
            "width": "2rem",
          }
        }
      >
        <ToggleIcon
          on={true}
        />
      </div>
      Desktop Modeler
    </li>
    <li
      aria-pressed={false}
      key="releasenotes/desktop-modeler/7"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "none",
          "borderBottom": "1px solid #ccc",
          "color": "#333",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 6rem",
        }
      }
    >
      <div
        style={
          Object {
            "height": "2rem",
            "left": "4.5rem",
            "position": "absolute",
            "width": "2rem",
          }
        }
      >
        <ToggleIcon
          on={false}
        />
      </div>
      7
    </li>
    <li
      aria-pressed={false}
      key="atd"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "none",
          "borderBottom": "1px solid #ccc",
          "color": "#333",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 3.5rem",
        }
      }
    >
      ATS Guide
    </li>
  </ul>
</Fragment>
`;

exports[`TreeViewMenu should render the level-1 nodes by default 1`] = `
<Fragment>
  <input
    aria-label="Type and search"
    onChange={[Function]}
    placeholder="Type and search"
    style={
      Object {
        "border": "none",
        "padding": "1rem 2rem",
        "width": "100%",
      }
    }
    type="search"
  />
  <ul
    style={
      Object {
        "borderTop": "1px solid #ccc",
        "listStyleType": "none",
        "paddingLeft": 0,
      }
    }
  >
    <li
      aria-pressed={false}
      key="releasenotes"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "none",
          "borderBottom": "1px solid #ccc",
          "color": "#333",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 3.5rem",
        }
      }
    >
      <div
        style={
          Object {
            "height": "2rem",
            "left": "2rem",
            "position": "absolute",
            "width": "2rem",
          }
        }
      >
        <ToggleIcon
          on={false}
        />
      </div>
      Release Notes
    </li>
    <li
      aria-pressed={false}
      key="atd"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "background": "none",
          "borderBottom": "1px solid #ccc",
          "color": "#333",
          "cursor": "pointer",
          "padding": " .75rem  1rem  .75rem 3.5rem",
        }
      }
    >
      ATS Guide
    </li>
  </ul>
</Fragment>
`;
